arch/arm/core/cortex_m/mpu/arm_mpu.c

```diff
@@ -29,7 +29,7 @@ static inline u32_t _get_region_attr(u32_t xn, u32_t ap, u32_t tex,
 				     u32_t srd, u32_t size)
 {
 	return ((xn << 28) | (ap) | (tex << 19) | (s << 18)
-		| (c << 17) | (b << 16) | (srd << 5) | (size));
+		| (c << 17) | (b << 16) | (srd << 8) | (size));
```

Comment:
Might be a good idea to assert hat these values are within their respective bounds

Reply:
Yeah some asserts would be good.  otherwise this looks fine.  And we probably need to use the #defines or make some for the bit shifts.  These might be present in the CMSIS if they aren't in the includes already defined in zephyr.

Reply:
I filed #6974 for these proposed enhancements, I'd like them to not block this fix